### ISSUE TITLE
[FR] Consolidations distributed traces and continues profiling

### ISSUE THREAD

**Issue by yaroslavkasatikov** [deepflowio/deepflow#8617]
### Search before asking

- [X] I had searched in the [issues](https://github.com/deepflowio/deepflow/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Hey team,
I would be wonderful if we can link tracing and profiling.
I found in profiler API "trace_id" field, but when I 'm checking it from ebpf profiler, the "trace_id" field is empty. 

### Use case

It can be very useful to collect all information regarding the trace. we can see not only traces (request etc),but check cpu/mem usage for all of them

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/deepflowio/deepflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by 1473371932** [comment]
Hi, DeepFlow itself doesn't passthrough `[trace id](https://www.deepflow.io/docs/features/distributed-tracing/auto-tracing/#autotracing)` in your business, you need to pass trace ids from your business itself for us to recognize them. This trace id can be called any name when you pass it in the business header, just configure [agent-group-config](https://github.com/deepflowio/deepflow/blob/main/server/agent_config/example.yaml#L299) via [deepflow-ctl](https://www.deepflow.io/docs/best-practice/agent-advanced-config/) and add the name.

**Comment by yaroslavkasatikov** [comment]
Hey @1473371932 ! Thank you for the respond. 
I was able to generate trace_id ( enabled opentelemery on nginx controller and it set the trace_id for me).
So I can see it in deepflow tracing now, but how can I link it to profiler?

**Comment by 1473371932** [comment]
> Hey @1473371932 ! Thank you for the respond. I was able to generate trace_id ( enabled opentelemery on nginx controller and it set the trace_id for me). So I can see it in deepflow tracing now, but how can I link it to profiler?

<img width="1903" alt="Snipaste_2024-12-10_03-54-07" src="https://github.com/user-attachments/assets/437a6553-7dd1-44c9-a126-6aea0e81bb16">

If your request passes through the nginx gateway, you need to generate an X-Request-ID in the gateway in order to correlate the request before and after the gateway, in the case of the trace id identification already configured above, you don't need to do any additional configuration, click on the request that has been identified in the above figure, it will automatically generate a flame map

**Comment by yaroslavkasatikov** [comment]
Yes, it works fine. But it's about tracing. My goal is link trace id to the Profiler. So I want to get profile for the application (for example - nginx ingress) in the moment when trace was captured. 
So here is the use case example.
I have Ollama after the ingress. And have many requests to Ollama from different sources. And on requests is implemented too long. 
So here is the issue investigation path which can be useful for me in this case:
1) I'm checking Deepflow continues tracing and looking up for long request's trace_id. 
2) I'm going to Continues profiling and looking for sample which is generated when process (ollama for the example) works too long. 
3) I'm trying to investigate which function( syscall ) was the reason of the request delay
